karapace/compatibility.py

```diff
@@ -135,6 +135,8 @@ def check_same_type(self, source, target):  # pylint: disable=no-self-use, too-m
             return isinstance(target.type, avro.schema.RecordSchema), False
         if isinstance(source.type, avro.schema.EnumSchema):
             return isinstance(target.type, avro.schema.EnumSchema), True
+        if isinstance(source.type, avro.schema.UnionSchema):
+            return isinstance(target.type, avro.schema.UnionSchema), True
```

Comment:
Union is a complex type tho, right? So should this not return False for the second value ?